hypervisor/arch/x86/configs/apl-mrb/ve820.c

```diff
@@ -7,33 +7,33 @@
 #include <e820.h>
 
 const struct e820_entry ve820_entry[E820_MAX_ENTRIES] = {
-	{	/* 0 to mptable */
+	{	/* usable RAM under 1MB */
 		.baseaddr = 0x0UL,
-		.length   = 0xEFFFFUL,
+		.length   = 0xF0000UL,		/* 960KB */
 		.type     = E820_TYPE_RAM
 	},
 
-	{	/* mptable 65536U */
-		.baseaddr = 0xF0000UL,
-		.length   = 0x10000UL,
+	{	/* mptable */
+		.baseaddr = 0xF0000UL,		/* 960KB */
+		.length   = 0x10000UL,		/* 16KB */
 		.type     = E820_TYPE_RESERVED
 	},
 
-	{	/* mptable to lowmem */
-		.baseaddr = 0x100000UL,
-		.length   = 0x1FF00000UL,
+	{	/* lowmem */
+		.baseaddr = 0x200000UL,		/* 2MB */
+		.length   = 0x1FF00000UL,	/* 511MB */
 		.type     = E820_TYPE_RAM
 	},
 
-	{	/* lowmem to PCI hole */
-		.baseaddr = 0x20000000UL,
-		.length   = 0xa0000000UL,
+	{	/* between lowmem and PCI hole */
+		.baseaddr = 0x20100000UL,	/* 513MB */
```

Comment:
why it starts from 513MB? should be 512MB.

Reply:
Because of the 1MB memory hole. So the 512MB memory is spread out: 0-1MB, 2MB-513MB. 
This looks weird. 

The alternative is to cut 1MB from the usable RAM and let it finish at 512MB. Then it makes the total available guest memory to be 511MB, still looks not normal.

Or we can keep the original solution to merge 1MB into the lowmem entry.

Reply:
No, it should be 0-960KB, 2MB-512MB.

Reply:
All the segment start and end address should be 2MB aligned, except the 0-1MB space

Reply:
OK, I can update accordingly.